metricbeat/module/elasticsearch/elasticsearch_integration_test.go

```diff
@@ -124,6 +122,92 @@ func TestData(t *testing.T) {
 	}
 }
 
+func TestXPackEnabled(t *testing.T) {
+	service := compose.EnsureUpWithTimeout(t, 300, "elasticsearch")
+	host := service.Host()
+
+	version, err := getElasticsearchVersion(host)
+	if err != nil {
+		t.Fatal("getting elasticsearch version", err)
+	}
+
+	setupTest(t, host, version)
+
+	metricSetToTypesMap := map[string][]string{
+		"ccr":            []string{"ccr_stats", "ccr_auto_follow_stats"},
+		"cluster_stats":  []string{"cluster_stats"},
+		"enrich":         []string{"enrich_coordinator_stats"},
+		"index_recovery": []string{"index_recovery"},
+		"index_summary":  []string{"indices_stats"},
+		"ml_job":         []string{"job_stats"},
+		"node_stats":     []string{"node_stats"},
+	}
+
+	config := getXPackConfig(host)
+
+	metricSets := mbtest.NewReportingMetricSetV2Errors(t, config)
+	for _, metricSet := range metricSets {
+		t.Run(metricSet.Name(), func(t *testing.T) {
+			checkSkip(t, metricSet.Name(), version)
+			events, errs := mbtest.ReportingFetchV2Error(metricSet)
+			assert.Empty(t, errs)
+			if !assert.NotEmpty(t, events) {
+				t.FailNow()
+			}
+
+			// Special case: the `index` metricset generates as many events
+			// as there are distinct indices in Elasticsearch
+			if metricSet.Name() == "index" {
+				numIndices, err := countIndices(host)
+				if !assert.NoError(t, err) {
+					t.FailNow()
+				}
+
+				if !assert.Len(t, events, numIndices) {
+					t.FailNow()
+				}
+
+				for _, event := range events {
+					assert.Equal(t, "index_stats", event.RootFields["type"])
+					assert.Regexp(t, `^.monitoring-es-\d-mb`, event.Index)
+				}
+
+				return
+			}
+
+			// Special case: the `shard` metricset generates as many events
+			// as there are distinct shards in Elasticsearch
+			if metricSet.Name() == "shard" {
+				numShards, err := countShards(host)
+				if !assert.NoError(t, err) {
```

Comment:
`require.NoError()`

Reply:
That's awesome, TIL!

I'll updated the usages of `assert.NoError` to `require.NoError` in this PR here and make a follow up PR to fix them in other tests.

Thanks!

Reply:
Follow up PR: https://github.com/elastic/beats/pull/15993.